packages/react-router-dev/cli/commands.ts

```diff
@@ -198,8 +209,14 @@ async function createClientEntry(
   return contents;
 }
 
-export async function typegen(root: string, flags: { watch: boolean }) {
-  root ??= process.cwd();
+export async function typegen(
+  root: string,
+  flags: {
+    watch?: boolean;
+    config?: string;
```

Comment:
@pcattori Just calling out that since the `--config` flag impacts the project root directory, I think it makes sense to support this in all CLI commands for consistency, even those that don't use the Vite config.